### DISCUSSION TITLE
Interactive Console

### DISCUSSION THREAD

**Discussion by j1o1h1n** [Textualize/textual#198] [Show and tell]
[Demo](https://asciinema.org/a/460467) | [Code](https://gist.github.com/j1o1h1n/2e8eea722a3a71d3190517a33569df57)

This is a demo of an interactive console in a Textual user interface.

* The console is activated with the tilde key
* Click to focus and expand
* The main app is available as local variable called “app”
* The demo shows
  * Update the header background colour
  * Call page down on the body
  * Asynchronously update the body contents

**Comment by willmcgugan** [comment]
This is awesome!

@darrenburns

**Reply by j1o1h1n** [reply]
Thanks!

**Reply by WaylonWalker** [reply]
This is UNREAL!

**Reply by davidbrochart** [reply]
@j1o1h1n I opened #202 which was heavily inspired by your gist.

**Reply by jeffwright13** [reply]
This is really cool :-)
